java-manta-client-unshaded/src/main/java/com/joyent/manta/client/crypto/AesCtrCipherDetails.java

```diff
@@ -121,6 +127,23 @@ public ByteRangeConversion translateByteRange(final long startInclusive, final l
                 ciphertextEndPositionInclusive, lengthOfPlaintextIncludingSkipBytes, startingBlockNumberInclusive);
     }
 
+    @Override
+    public AlgorithmParameterSpec getEncryptionParameterSpec(final byte[] iv, final long position) {
+        Validate.notNull(iv, "Initialization vector must not be null");
+        Validate.isTrue(iv.length == getIVLengthInBytes(),
+                "Initialization vector has the wrong byte count [%d] "
+                        + "expected [%d] bytes", iv.length, getIVLengthInBytes());
+
+        final int blockSize = getBlockSizeInBytes();
+        final long startingBlock = position / blockSize;
+        final byte[] counter = ByteBuffer.allocate(16).putLong(startingBlock).array();
```

Comment:
`counter` remains used in this function.

Reply:
Good catch. I removed that.